collectors/python.d.plugin/vnstat/vnstat.chart.py

```diff
@@ -0,0 +1,731 @@
+                            (
+                                "years_average_"
+                                + interface
+                                + "_year"
+                                + str(hash(year[0]))[1:]
+                            ),
+                            year[0],
+                            "absolute",
+                            1,
+                            1,
+                        ]
+                        charts["years_average_" + interface_or_chart]["lines"].append(
+                            new_line
+                        )
+
+                        self.add_chart_dimension("years_average_" + interface_or_chart, new_line)
+
+                        data[
+                            (
+                                "years_average_"
+                                + interface
+                                + "_year"
+                                + str(hash(year[0]))[1:]
+                            )
+                        ] = average_amount
+
+                    if (data_representation == 0 or data_representation == 1):
+                        new_line = [
+                            (
+                                "years_total_"
+                                + interface
+                                + "_year"
+                                + str(hash(year[0]))[1:]
+                            ),
+                            year[0],
+                            "absolute",
+                            1,
+                            1,
+                        ]
+                        charts["years_total_" + interface_or_chart]["lines"].append(
+                            new_line
+                        )
+
+                        self.add_chart_dimension("years_total_" + interface_or_chart, new_line)
+
+                        data[
+                            (
+                                "years_total_"
+                                + interface
+                                + "_year"
+                                + str(hash(year[0]))[1:]
+                            )
+                        ] = year[1]
+
+            charts["totals_" + interface_or_chart] = {
+                "options": [
+                    None,
+                    "Total RX vs. Total TX",
+                    "B",
+                    "total bandwidth " + interface,
+                    "vnstat.totals_" + interface_or_chart,
+                    "stacked",
+                ],
+                "lines": [
+                    ["totals_" + interface + "_rx", "rx", "absolute", 1, 1],
+                    ["totals_" + interface + "_tx", "tx", "absolute", 1, 1],
+                ],
+            }
+
+            order.append("totals_" + interface_or_chart)
+
+            data["totals_" + interface + "_rx"] = data_points[index][0]
+            data["totals_" + interface + "_tx"] = data_points[index][1]
+
+            index += 1
+
+        return charts, order, data
```

Comment:
This 500+ lines function is very hard to read, can we split it into several functions? It is hard to suggest something because of size and formatting.


```pyhon
        try:
            hours_limit = int(self.configuration.get("hours_limit", 0))
            days_limit = int(self.configuration.get("days_limit", 0))
            months_limit = int(self.configuration.get("months_limit", 0))
            years_limit = int(self.configuration.get("years_limit", 0))
        except ValueError:
            hours_limit = 0
            days_limit = 0
            months_limit = 0
            years_limit = 0
```

all this could be done once in `__init__`

Reply:
Okay, that section of code has been moved to `__init__`

As for splitting-up the chart-generation function, I think doing so might make it even messier. Would you be opposed to me adding comments to section pieces off, to make it more readable?